File: bubblesort.py
def countSwaps(a):
    n = len(a)
    cnt = 0
    for i in range(n):
        for j in range(n-1):
            if (a[j] > a[j + 1]):
                temp = a[j]
                a[j] = a[j + 1]
                a[j + 1] = temp
                cnt += 1
    print('Array is sorted in {} swaps.'.format(cnt))
    print('First Element: {}'.format(a[0]))
    print('Last Element: {}'.format(a[n-1]))


countSwaps([3, 2, 1])

File: bubblesort.js
function countSwaps(a) {
  let n = a.length,
    cnt = 0;
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n - 1; j++) {
      if (a[j] > a[j + 1]) {
        // let temp = a[j]; a[j] = a[j + 1]; a[j + 1] = temp;
        [a[j], a[j + 1]] = [a[j + 1], a[j]]; //ES6 way of swap
        cnt++;
      }
    }
  }
  return a;
}

console.log(countSwaps([3, 2, 1, 1, 2, 2, 3, 5, 7, 8]));
